### DISCUSSION TITLE
[04.아키텍처] 버퍼 풀과 리두 로그

### DISCUSSION THREAD

**Discussion by Youngcircle-kim** [Learning-Is-Vital-In-Development/23-6-RealMySQL#13] [General]
p.111 버퍼 풀과 리두 로그 둘의 관계가 잘 모르겠습니다..

**Comment by dojinyou** [comment]
버퍼 풀은 스토리지 엔진이 디스크에 작업하기 전에 데이터를 미리 변경하는 캐시 역할을 하고 있습니다. 이러한 캐시 데이터들을 보장하기 위해서는 리두 로그가 해당 행위를 보장하고 있어야 합니다. 
리두 로그 전체 파일의 크기가 클 수록 많은 데이터를 버퍼링할 수 있으나 실제로 디스크에 쓰기 연산이 발생할 때 문제가 생기거나 많은 부하를 가할 수 있습니다. 반대로 리두 로그 전체 파일의 크기가 작다면 버퍼 풀의 크기와 무관하게 실제로 버퍼링 되는 더티 페이지의 사이즈가 제한되어 메모리를 효과적으로 사용할 수 없고 잦은 디스크 쓰기가 발생합니다. 
따라서 서비스의 사용형태에 맞는 적절한 사이즈를 찾아 이용하는 것이 필요합니다.
